src/main/scala/tech/cryptonomic/conseil/generic/chain/DataTypes.scala

```diff
@@ -19,15 +21,30 @@ object DataTypes {
   import cats.implicits._
   import io.scalaland.chimney.dsl._
 
+
   /** Type representing Map[String, Option[Any]] for query response */
   type QueryResponse = Map[String, Option[Any]]
-  /** Method checks if type can be aggregated */
-  lazy val canBeAggregated: DataType => Boolean = Set(DataType.Decimal, DataType.Int, DataType.LargeInt, DataType.DateTime)
   /** Default value of limit parameter */
   val defaultLimitValue: Int = 10000
   /** Max value of limit parameter */
   val maxLimitValue: Int = 100000
 
+  /** Replaces timestamp represented as Long in predicates with one understood by the SQL */
+  private def replaceTimestampInPredicates(entity: String, query: Query, tezosPlatformDiscoveryOperations: TezosPlatformDiscoveryOperations)
+    (implicit executionContext: ExecutionContext): Future[Query] = {
+    query.predicates.map { predicate =>
+      tezosPlatformDiscoveryOperations.getTableAttributesWithoutUpdatingCache(entity).map { maybeAttributes =>
+        maybeAttributes.flatMap { attributes =>
+          attributes.find(_.name == predicate.field).map {
+            case attribute if attribute.dataType == DataType.DateTime =>
+              predicate.copy(set = predicate.set.map(x => new Timestamp(x.toString.toLong).toInstant.toString))
```

Comment:
Are we sure that in both places we use `.toString`, the result would always be the one that we have now?

- `x.toString` seems quite plausible
- I'm not too keen on the `.toInstant.toString`, I'm worried that 
  - the default toString of `Instant` might change in the future
  - we're not considering timezone-independence of this code (in particular wrt. the database's own TZ)

for the latter I can think of using an explicit iso format defined in java.time that would
- make the assumptions and conventions clear
- take the timezone into account

Reply:
ok, changed

Reply:
I outlined a different way to do that, using java.time apis